### PR TITLE
adding in missing dependencies for plugins

### LINKED ISSUE #86: the groovy-postbuild plugin was installed but jenkins didn't register it
The package appeared to be installed on the filesystem, but jenkins listed it as "available" and many jobs were crashing because it was not installed. 

The work around is to install it via the gui.

### PR DESCRIPTION
**Pull request by tfoote**
determined as all plugins manually installed by the update center for groovy-postbuild plugin not already listed

Fixes #86

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please keep all plugins in alphabetical order and use a consistent comment (`required by ...`) in front of every plugin which is only added as a dependency.

**Comment by tfoote** [comment]
I rewrote it to be alphabetized, consistently labeled and used a more compact format for easier reading.

File: master/manifests/site.pp [hunk 1]
```diff
@@ -76,31 +76,16 @@
...
-
-jenkins::plugin {
-  'build-timeout': ;
-}
-
-jenkins::plugin {
-  'collapsing-console-sections': ;
-}
+# required by groovy-postbuild
+jenkins::plugin { 'ace-editor': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: Where did you find this? I don't see it anywhere in here: https://github.com/jenkinsci/groovy-postbuild-plugin/blob/master/pom.xml

File: master/manifests/site.pp [hunk 2]
```diff
@@ -160,68 +126,29 @@
...
-
-jenkins::plugin {
-  'jobrequeue': ;
-}
-
+jenkins::plugin { 'github': ; }
+jenkins::plugin { 'github-api': ; }
+jenkins::plugin { 'greenballs': ; }
+jenkins::plugin { 'groovy': ; }
+jenkins::plugin { 'groovy-postbuild': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: Isn't `workflow-job` still missing?

File: master/manifests/site.pp [hunk 3]
```diff
@@ -244,59 +169,58 @@
...
-}
-
-jenkins::plugin {
-  'swarm':
+jenkins::plugin { 'script-security': ; }
+# required by groovy-postbuild
+# required by groovy-postbuild
+jenkins::plugin { 'durable-task': ; }
+# required by groovy-postbuild
+jenkins::plugin { 'ssh-agent': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: We need `ssh-agent` for sure but where did you find that it is a requirement for `groovy-postbuild`?

File: master/manifests/site.pp [hunk 4]
```diff
@@ -244,59 +169,58 @@
...
+jenkins::plugin { 'subversion': ; }
+jenkins::plugin { 'systemloadaverage-monitor': ; }
+jenkins::plugin { 'swarm':
     require => Package['wget'];
 }
+jenkins::plugin { 'timestamper': ; }
+# required by build-timeout
+jenkins::plugin { 'token-macro': ; }
+# required by groovy-postbuild
+jenkins::plugin { 'workflow-api': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: Where did you find this? I don't see it anywhere in here: https://github.com/jenkinsci/groovy-postbuild-plugin/blob/master/pom.xml

File: master/manifests/site.pp [hunk 5]
```diff
@@ -244,59 +169,58 @@
...
+# needed for ssh-agent
+jenkins::plugin { 'workflow-cps': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: This is needed by `groovy-postbuild` but not by `ssh-agent`?

File: master/manifests/site.pp [hunk 6]
```diff
@@ -244,59 +169,58 @@
...
+# required by groovy-postbuild
+jenkins::plugin { 'workflow-scm-step': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: Where did you find this? I don't see it anywhere in here: https://github.com/jenkinsci/groovy-postbuild-plugin/blob/master/pom.xml

File: master/manifests/site.pp [hunk 7]
```diff
@@ -244,59 +169,58 @@
...
+# required by groovy-postbuild
+jenkins::plugin { 'workflow-support': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: Where did you find this? I don't see it anywhere in here: https://github.com/jenkinsci/groovy-postbuild-plugin/blob/master/pom.xml

File: master/manifests/site.pp [hunk 8]
```diff
@@ -244,59 +169,58 @@
...
+jenkins::plugin { 'workflow-step-api': ; }
```
Comment (dirk-thomas) [inline master/manifests/site.pp]: Where did you find this?

**Comment by dirk-thomas** [comment]
Due to the many comments and unreadable diff I have created #101.

**Comment by tfoote** [comment]
We need to install all the recursive dependencies. I found the list by executing the install of the desired missing plugin (groovy-postbuild) and watching for newly installed plugins on the filesystem. I did not take the time to trace every one down exactly.

You can test by provisioning an instance following the standard procedures and seeing if it loads the plugin

File: master/manifests/site.pp (hunk 2 above)
Comment (tfoote) [inline master/manifests/site.pp]: workflow-job is not on the machine

Comment (dirk-thomas) [inline master/manifests/site.pp]: It's only a test dependency: https://github.com/jenkinsci/groovy-postbuild-plugin/blob/7df1063ef6e1bc786cff7326fec08305e067aaf4/pom.xml#L91

I removed it from #101 again.

**Comment by tfoote** [comment]
You can see the files being created by the update center
```
-rw-r----- 1 jenkins jenkins  165K Jan  8 22:01 script-security.jpi
-rw-r----- 1 jenkins jenkins     0 Jan  8 22:01 script-security.jpi.pinned
-rw-r----- 1 jenkins jenkins     0 Jan  8 22:01 credentials.jpi.pinned
-rw-r----- 1 jenkins jenkins  416K Jan  8 22:01 credentials.jpi
-rw-r----- 1 jenkins jenkins     0 Jan  8 22:01 ssh-credentials.jpi.pinned
-rw-r----- 1 jenkins jenkins   70K Jan  8 22:01 ssh-credentials.jpi
-rw-r----- 1 jenkins jenkins     0 Jan  8 22:01 junit.jpi.pinned
-rw-r----- 1 jenkins jenkins  326K Jan  8 22:01 junit.jpi
-rw-r----- 1 jenkins jenkins     0 Jan  8 22:01 mailer.jpi.pinned
-rw-r----- 1 jenkins jenkins  113K Jan  8 22:01 mailer.jpi
drwxr-x--- 6 jenkins jenkins  4.0K Jan  8 22:03 credentials
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:03 script-security
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:03 junit
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:03 ssh-credentials
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:03 mailer
-rw-r--r-- 1 jenkins jenkins  4.5M Jan  8 22:26 subversion.bak
-rw-r----- 1 jenkins jenkins   44K Jan  8 22:32 workflow-step-api.jpi
-rw-r----- 1 jenkins jenkins     0 Jan  8 22:32 subversion.jpi.pinned
-rw-r----- 1 jenkins jenkins  6.8M Jan  8 22:32 subversion.jpi
-rw-r----- 1 jenkins jenkins  230K Jan  8 22:32 workflow-scm-step.jpi
-rw-r----- 1 jenkins jenkins   46K Jan  8 22:32 workflow-api.jpi
-rw-r----- 1 jenkins jenkins  6.7M Jan  8 22:32 jquery-detached.jpi
-rw-r----- 1 jenkins jenkins   31K Jan  8 22:32 durable-task.jpi
-rw-r----- 1 jenkins jenkins  423K Jan  8 22:32 workflow-support.jpi
-rw-r----- 1 jenkins jenkins  4.1M Jan  8 22:32 ace-editor.jpi
-rw-r----- 1 jenkins jenkins  1.9M Jan  8 22:32 workflow-cps.jpi
-rw-r----- 1 jenkins jenkins   34K Jan  8 22:32 groovy-postbuild.jpi
drwxr-x--- 7 jenkins jenkins  4.0K Jan  8 22:33 ace-editor
drwxr-x--- 5 jenkins jenkins  4.0K Jan  8 22:33 workflow-support
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:33 workflow-step-api
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:33 workflow-scm-step
drwxr-x--- 5 jenkins jenkins  4.0K Jan  8 22:33 groovy-postbuild
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:33 durable-task
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:33 workflow-api
drwxr-x--- 5 jenkins jenkins  4.0K Jan  8 22:33 workflow-cps
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:33 subversion
drwxr-x--- 6 jenkins jenkins  4.0K Jan  8 22:33 jquery-detached
-rw-r--r-- 1 jenkins jenkins  928K Jan  8 22:36 windows-slaves.jpi
-rw-r--r-- 1 jenkins jenkins   49K Jan  8 22:36 translation.jpi
-rw-r--r-- 1 jenkins jenkins   63K Jan  8 22:36 ssh-slaves.jpi
-rw-r--r-- 1 jenkins jenkins 1007K Jan  8 22:36 pam-auth.jpi
-rw-r--r-- 1 jenkins jenkins   11M Jan  8 22:36 maven-plugin.jpi
-rw-r--r-- 1 jenkins jenkins  234K Jan  8 22:36 matrix-project.jpi
-rw-r--r-- 1 jenkins jenkins  100K Jan  8 22:36 ldap.jpi
-rw-r--r-- 1 jenkins jenkins   38K Jan  8 22:36 javadoc.jpi
-rw-r--r-- 1 jenkins jenkins   55K Jan  8 22:36 external-monitor-job.jpi
-rw-r--r-- 1 jenkins jenkins  908K Jan  8 22:36 cvs.jpi
-rw-r--r-- 1 jenkins jenkins   89K Jan  8 22:36 ant.jpi
-rw-r--r-- 1 jenkins jenkins  110K Jan  8 22:36 antisamy-markup-formatter.jpi
-rw-r--r-- 1 jenkins jenkins   78K Jan  8 22:36 matrix-auth.jpi
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:36 ldap
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:36 pam-auth
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:36 antisamy-markup-formatter
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:36 external-monitor-job
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:36 maven-plugin
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:36 translation
drwxr-x--- 5 jenkins jenkins  4.0K Jan  8 22:36 matrix-project
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:36 windows-slaves
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:37 javadoc
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:37 ant
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:37 matrix-auth
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:37 ssh-slaves
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:37 cvs
-rw-r----- 1 jenkins jenkins   15K Jan  8 22:45 plain-credentials.jpi
-rw-r----- 1 jenkins jenkins  3.0M Jan  8 22:45 github.jpi
-rw-r----- 1 jenkins jenkins  361K Jan  8 22:45 ghprb.jpi
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:53 ghprb
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:53 github
drwxr-x--- 4 jenkins jenkins  4.0K Jan  8 22:53 plain-credentials
-rw-r----- 1 jenkins jenkins  670K Jan 11 10:47 github-oauth.jpi
drwxr-x--- 5 jenkins jenkins  4.0K Jan 11 10:47 github-oauth
```

The jpi's are the files downloaded by the update center.  When I shuffled there were a few of the reason lines disassociated. The last jpi is the one I requested to be installed, and the rest of dependencies filled in by jenkins plugin updater.

File: master/manifests/site.pp (hunk 4 above)
Comment (dirk-thomas) [inline master/manifests/site.pp]: Required by `workflow-cps`.

File: master/manifests/site.pp (hunk 6 above)
Comment (dirk-thomas) [inline master/manifests/site.pp]: Required by `workflow-cps`.

File: master/manifests/site.pp (hunk 7 above)
Comment (dirk-thomas) [inline master/manifests/site.pp]: Required by `workflow-cps`.

File: master/manifests/site.pp (hunk 8 above)
Comment (dirk-thomas) [inline master/manifests/site.pp]: Required by `ssh-agent` and `workflow-cps`.

**Comment by dirk-thomas** [comment]
The problem is not the list of plugins itself. The puppet file also documents why they are needed. By now I have manually traced all dependencies and documented them accordingly in #101 except `ace-editor`.

File: master/manifests/site.pp (hunk 1 above)
Comment (dirk-thomas) [inline master/manifests/site.pp]: Required by `workflow-cps`.

**Comment by dirk-thomas** [comment]
I finally found the `ace-editor` dependency.

I would vote for closing this and merging #101.

**Comment by tfoote** [comment]
Close in favor of #101

**Pull request closed without merging**